pkg/autoscaling/prometheus.go

```diff
@@ -164,3 +179,35 @@ func buildCPUUsagePromQL(options *QueryOptions) (string, error) {
 	query := fmt.Sprintf(pattern, options.duration.String())
 	return query, nil
 }
+
+// this function assumes that addr is already a valid resolvable address
+// returns in format "podname_namespace"
+func getInstanceNameFromAddress(addr string) (string, error) {
+	// In K8s, a StatefulSet pod address is composed of pod-name.peer-svc.namespace.svc:port
+	// Extract the hostname part without port
+	hostname := addr
+	portColonIdx := strings.LastIndex(addr, ":")
+	if portColonIdx >= 0 {
+		hostname = addr[:portColonIdx]
+	}
+
+	// Just to make sure it is not an IP address
+	ip := net.ParseIP(hostname)
+	if ip != nil {
+		// Hostname is an IP address, return the whole address
+		return "", errors.Errorf("address %s is an ip address", addr)
+	}
+
+	parts := strings.Split(hostname, ".")
+	if len(parts) < 4 {
+		return "", errors.Errorf("address %s has less than 4 parts", addr)
```

Comment:
This log is poor readability. I prefer to output the format we expect and the actual result